src/main/java/com/sourcegraph/cody/CodyFocusChangeListener.java

```diff
@@ -45,17 +45,17 @@ public void focusGained(@NotNull Editor editor) {
       return;
     }
 
-    CodyAgentService.applyAgentOnBackgroundThread(
-        project,
-        agent -> {
-          try {
-            // TODO: This is bad but needed to avoid race with file context of commands executed
-            // through context menu
-            Thread.sleep(100);
-          } catch (InterruptedException ignored) {
-          }
-          agent.getServer().textDocumentDidFocus(TextDocument.fromPath(file.getPath()));
-        });
+    CodyAgentService.withAgent(project)
+        .thenAccept(
+            agent -> {
+              try {
+                // TODO: This is bad but needed to avoid race with file context of commands executed
```

Comment:
oh, this TODO was here already. ok then. I wonder if we have any idea how to fix it some day 🤔